### ISSUE TITLE
Optionally disallow access modifier override on extensions

### ISSUE THREAD

**Issue by masters3d** [realm/SwiftLint#1457]
Instead of
```swift
private extension SomeType {
    func someFuncA() {}
}
```
Prefer
```swift
extension SomeType {
    fileprivate func someFuncA() {}
}
```

**Comment by masters3d** [comment]
Find all the occurrences of `extension` in the structure: 
 ``"key.kind" : "source.lang.swift.decl.extension"``
then we could find out if there is a syntax type before the `extension`:
``"type" : "source.lang.swift.syntaxtype.attribute.builtin"``
then we check if the built in attribute is any of the access modifier;
If true, then we can mark the violation.
